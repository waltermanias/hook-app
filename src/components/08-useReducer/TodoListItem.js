import React from "react";
import PropTypes from "prop-types";

const TodoListItem = ({ index, todo, handleToggle, handleDelete }) => {
  return (
    <li className="list-group-item" key={todo.id}>
      <p
        onClick={() => handleToggle(todo.id)}
        className={`${todo.done ? "complete" : ""}`}
      >{`${index + 1}. ${todo.desc}`}</p>
      <button className="btn btn-danger" onClick={() => handleDelete(todo.id)}>
        Borrar
      </button>
    </li>
  );
};

TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default TodoListItem;
