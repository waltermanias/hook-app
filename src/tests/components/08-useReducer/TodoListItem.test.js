import { shallow } from "enzyme";
import React from "react";

import TodoListItem from "../../../components/08-useReducer/TodoListItem";

describe("TodoListItem", () => {
  it("should render the component", () => {
    const wrapper = shallow(
      <TodoListItem
        index={1}
        todo={{ done: false, desc: "test-desc", id: 1 }}
        handleToggle={() => {}}
        handleDelete={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('p').text().trim()).toBe("2. test-desc")
  });

  it("should call handleToggle with params", () => {
    const handleToggle = jest.fn();

    const wrapper = shallow(
      <TodoListItem
        index={1}
        todo={{ done: false, desc: "test-desc", id: 1 }}
        handleToggle={handleToggle}
        handleDelete={() => {}}
      />
    );

    wrapper.find("p").simulate("click");

    expect(handleToggle).toHaveBeenCalledWith(1);
  });

  it("should call handleDelete with params", () => {
    const handleDelete = jest.fn();

    const wrapper = shallow(
      <TodoListItem
        index={1}
        todo={{ done: false, desc: "test-desc", id: 1 }}
        handleToggle={() => {}}
        handleDelete={handleDelete}
      />
    );

    wrapper.find("button").simulate("click");

    expect(handleDelete).toHaveBeenCalledWith(1);
  });
});
