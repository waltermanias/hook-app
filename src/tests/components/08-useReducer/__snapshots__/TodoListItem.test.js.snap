// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem should render the component 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className=""
    onClick={[Function]}
  >
    2. test-desc
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
